### DISCUSSION TITLE
getPatchAsImage edge cases outside image level size

### DISCUSSION THREAD

**Discussion by mhoibo** [FAST-Imaging/FAST#177] [Q&A] [answered]
I am importing annotations as tiffs and corresponding vsis with FAST. This results in slightly different sizes of the images. I cannot see a difference visually, does this mean that is is a padding difference? This also results in warnings when using getPatchAsImage on edge cases. Both the offset + size exceeds level size and a tiff warning about row/col out of range. The tiff warning still enables me to get the patch, and I cannot see a difference in the annotated image and the vsi, thus I assume the patch is only padded? Is this correct? Is this the same for the offset + size exceeds level size warning?

**Comment by andreped** [comment]
@mhoibo Might be that you deleted your initial questions during edit, because I am unable to see anything now. From the dropdown **edited**, I can see your original reply. You might want to revert that edit.

**Comment by smistad** [comment, marked as answer]
I am assuming this concerns Olympus CellSense VSI images. We have observed that the size reported in QuPath is not the same as the size reported by FAST when opening the .vsi files. This is probably because the size of these images can be calculated in many differnt ways as the format doesn't follow a strict image pyramid format, and some tiles might be present in one magnification and not in another.

I think the most important thing is whether your exported annotations overlap with the .vsi image when importing them in FAST. This you can verify visually by rendering them on top of eachother.

**Reply by mhoibo** [reply]
Thank you!

**Reply by andreped** [reply]
> Both the offset + size exceeds level size and a tiff warning about row/col out of range.

The cause to why this happened way more often than previously, was due to a scaling difference between the HE and CK not performed correctly, when attempting to extract full TMA pairs. After doing that, everything seem to be working as intended.
